stdlib/public/Atomics/UnsafeAtomicLazyReference.swift

```diff
@@ -0,0 +1,135 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2020 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import Swift
+
+/// An atomic optional strong reference that can be set (initialized) exactly
+/// once, but read many times.
+@available(macOS 9999, iOS 9999, watchOS 9999, tvOS 9999, *)
+@frozen
+public struct UnsafeAtomicLazyReference<Instance: AnyObject> {
+  public typealias Value = Instance?
```

Comment:
It might be clearer to remove this typealias, and use `Instance?` directly.

(`Instance` and `Instance?` are already used in the store/load methods.)

Reply:
`Value` is part of the new unsafe API pattern for high-performance synchronization primitives. This is pure 100% ceremony, but it helps establishing / highlighting patterns for such types.